src/WebJobs.Script.WebHost/TokenExpiration/TokenExpirationService.cs

```diff
@@ -0,0 +1,184 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Reflection.PortableExecutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.Storage;
+using Microsoft.Azure.Storage.Auth;
+using Microsoft.Azure.WebJobs.Script.Diagnostics;
+using Microsoft.Azure.WebJobs.Script.Properties;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost.TokenExpiration
+{
+    internal class TokenExpirationService : IHostedService, IDisposable
+    {
+        private readonly IEnvironment _environment;
+        private readonly ILoggerFactory _loggerFactory;
+        private readonly IOptionsMonitor<StandbyOptions> _standbyOptionsMonitor;
+        private readonly WebJobsScriptHostService _scriptHost;
+        private CancellationTokenSource _cancellationTokenSource;
+        private Task _analysisTask;
+        private bool _disposed;
+        private bool _analysisScheduled;
+        private ILogger _logger;
+
+        public TokenExpirationService(IEnvironment environment, WebJobsScriptHostService scriptHost, ILoggerFactory loggerFactory, IOptionsMonitor<StandbyOptions> standbyOptionsMonitor)
+        {
+            _environment = environment;
+            _scriptHost = scriptHost;
+            _loggerFactory = loggerFactory;
+            _standbyOptionsMonitor = standbyOptionsMonitor;
+            _logger = _loggerFactory.CreateLogger<TokenExpirationService>();
+        }
+
+        public Task StartAsync(CancellationToken cancellationToken)
+        {
+            try
+            {
+                if (!_environment.IsCoreTools())
```

Comment:
nit: Consider inverting if to reduce nesting (of `{`) which helps with readability.

Reply:
can you elaborate? I see there's if statement for checking if the env is not core tools, but not sure how I'd invert that to reduce the {} ?

Reply:
ohhh i just read the next comment. I'll use continue :)

Reply:
I think he means to do something like this so everything isn't nested:

```csharp
if (_environment.IsCoreTools())
{
    return Task.CompletedTask;
}
```